### PR TITLE
Framework: Replace click-outside by react-click-outside

### PR DESCRIPTION
**Pull request by ockham**
Second stab at #5181.

The only remaining component to use `click-outside` is `Popover`, which this PR modifies to use `react-click-outside` instead. `click-outside` is then removed from `package.json`. `react-click-outside` works much better with React in general, and with server-side rendering in particular.

However, we cannot drop [our stubbing](https://github.com/Automattic/wp-calypso/blob/fffa03b3406ec5421d87e949221e8b0a09bdbd87/webpack.config.node.js#L108) quite yet, since `Popover` [still makes use](https://github.com/Automattic/wp-calypso/blob/fffa03b3406ec5421d87e949221e8b0a09bdbd87/client/components/popover/util.js) of `window` directly. In the long run, maybe we should use a ready-made React tooltip component after all (see #4072). Or we guard that code with `typeof 'window' !== 'undefined'` checks.

To test:
* Verify that `Popovers` still work as before -- especially that they disappear when clicking outside 😄

### CODE REVIEW

**Comment by ockham** [comment]
Pinging for reviews 😛

File: client/components/popover/index.jsx [hunk 1]
```diff
@@ -166,28 +165,8 @@ class Popover extends Component {
...
-	}
-
-	unbindClickoutHandler() {
-		if ( this._clickoutHandlerReference ) {
-			this.debug( 'unbinding `clickout` listener ...' );
-			this._clickoutHandlerReference();
-			this._clickoutHandlerReference = null;
-		}
+	handleClickOutside( event ) {
+		this.onClickout( event );
```
Comment (retrofox) [inline client/components/popover/index.jsx]: I wonder if we could straightly remove `onClickout` method and declare the whole behavior into the `handleClickOutside` method body. We should remove the `onClickoout` binding as well.

```js
  handleClickOutside( event ) {
    let shouldClose = (
    // ...
```

Comment (ockham) [inline client/components/popover/index.jsx]: Yep, that makes a lot of sense. Fixed in 8305c2a.

**Comment by ockham** [comment]
Thanks for reviewing @retrofox!

**Pull request merged**